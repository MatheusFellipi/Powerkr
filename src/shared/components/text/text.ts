import { styled, Text as T } from "tamagui";

export const TextComponent = styled(T, {
  fontSize: 14,
  fontFamily: "Inter",
  fontWeight: "400",
  variants: {
    pin: {
      title: {
        color: "#fff",
        fontSize: 20
      },
      subText: {
        color: "#CDD5DF"
      },
      description: {
        color: "#9AA4B2"
      }
    }
  } as const
});
