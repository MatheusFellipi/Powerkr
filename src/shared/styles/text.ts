import { styled, Text } from "tamagui"; // or '@tamagui/core'
export const Title = styled(Text, {
  variants: {
    pin: {
      top: {
        position: "absolute",
        top: 0
      }
    },
    centered: {
      true: {
        alignItems: "center",
        justifyContent: "center"
      }
    },
    size: {
      "...size": (size, { tokens }) => {
        return {
          width: tokens.size[size] ?? size,
          height: tokens.size[size] ?? size
        };
      }
    }
  } as const
});
