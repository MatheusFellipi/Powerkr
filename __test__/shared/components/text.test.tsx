import { TextComponent } from "@shared/components/text/text";
import { render, screen } from "@testing-library/react-native";
import { TamaguiProvider } from "tamagui";

import config from "../../../tamagui.config";

describe("<App />", () => {
  it("renders correctly", () => {
    const tree = render(
      <TamaguiProvider config={config}>
        <TextComponent pin="description">Test</TextComponent>
      </TamaguiProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
    expect(screen.getByText("Test")).toBeDefined();
  });
  it("renders correctly pin title", () => {
    const tree = render(
      <TamaguiProvider config={config}>
        <TextComponent pin="title">Test</TextComponent>
      </TamaguiProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
    expect(screen.getByText("Test")).toBeDefined();
  });
  it("renders correctly pin sub text", () => {
    const tree = render(
      <TamaguiProvider config={config}>
        <TextComponent pin="subText">Test</TextComponent>
      </TamaguiProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
    expect(screen.getByText("Test")).toBeDefined();
  });
});
