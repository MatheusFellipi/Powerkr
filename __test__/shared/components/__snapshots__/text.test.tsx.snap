// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<Text
  fontFamily="Inter"
  style={
    {
      "color": "#9AA4B2",
      "fontFamily": "Inter",
      "fontSize": 14,
    }
  }
  suppressHighlighting={true}
>
  Test
</Text>
`;

exports[`<App /> renders correctly pin sub text 1`] = `
<Text
  fontFamily="Inter"
  style={
    {
      "color": "#CDD5DF",
      "fontFamily": "Inter",
      "fontSize": 14,
    }
  }
  suppressHighlighting={true}
>
  Test
</Text>
`;

exports[`<App /> renders correctly pin title 1`] = `
<Text
  fontFamily="Inter"
  style={
    {
      "color": "#fff",
      "fontFamily": "Inter",
      "fontSize": 20,
    }
  }
  suppressHighlighting={true}
>
  Test
</Text>
`;
